src/EVM/SymExec.hs

```diff
@@ -674,34 +673,33 @@ verify solvers opts preState maybepost = do
 
   exprInter <- interpret (Fetch.oracle solvers opts.rpcInfo) opts.iterConf preState runExpr
   when conf.dumpExprs $ liftIO $ T.writeFile "unsimplified.expr" (formatExpr exprInter)
-  liftIO $ do
-    when conf.debug $ putStrLn "   Simplifying expression"
+  do
+    when conf.debug $ liftIO $ putStrLn "   Simplifying expression"
     let expr = if opts.simp then (Expr.simplify exprInter) else exprInter
-    when conf.dumpExprs $ T.writeFile "simplified.expr" (formatExpr expr)
-    when conf.dumpExprs $ T.writeFile "simplified-conc.expr" (formatExpr $ Expr.simplify $ mapExpr Expr.concKeccakOnePass expr)
+    when conf.dumpExprs $ liftIO $ T.writeFile "simplified.expr" (formatExpr expr)
+    when conf.dumpExprs $ liftIO $ T.writeFile "simplified-conc.expr" (formatExpr $ Expr.simplify $ mapExpr Expr.concKeccakOnePass expr)
     let flattened = flattenExpr expr
-    when conf.debug $ do
+    when conf.debug $ liftIO $ do
```

Comment:
This seems unrelated to the main point of the PR.
To me it seems better to write `liftIO` once. What is the problem with that?